web/test/import.test.ts

```diff
@@ -0,0 +1,461 @@
+      type: 'JOIN_PROJECT_CELL_ID',
+      payload: mockCellIdString,
+    })
+
+    expect(store.getState).toHaveBeenCalledTimes(1)
+
+    expect(onStep).toHaveBeenCalledTimes(
+      1 + sampleGoodDataExport.projects.length
+    )
+  })
+
+  it('throws error with invalid json', async () => {
+    mockMigrationData = 'invalid json'
+    try {
+      await internalImportProjectsData(
+        profilesZomeApi,
+        projectsZomeApi,
+        installProjectAppAndImport,
+        installProjectApp,
+        store,
+        mockMigrationData,
+        onStep
+      )
+    } catch (e) {
+      expect(e).toBeInstanceOf(SyntaxError)
+      expect(e.message).toBe('Unexpected token i in JSON at position 0')
+    }
+
+    mockMigrationData = null
+    try {
+      await internalImportProjectsData(
+        profilesZomeApi,
+        projectsZomeApi,
+        installProjectAppAndImport,
+        installProjectApp,
+        store,
+        mockMigrationData,
+        onStep
+      )
+    } catch (e) {
+      expect(e).toBeInstanceOf(TypeError)
+      expect(e.message).toBe(
+        "Cannot read properties of null (reading 'projects')"
+      )
+    }
+  })
+})
+
+describe('installProjectAppAndImport()', () => {
+  const agentAddress = 'testAgentAddress'
+  it('installs DNA and imports project into new cell', async () => {
+    await internalInstallProjectAppAndImport(
+      agentAddress,
+      sampleGoodDataExport.projects[0],
+      sampleGoodDataExport.projects[0].projectMeta.passphrase,
+      store.dispatch,
+      installProjectApp,
+      createActionHashMapAndImportProjectData,
+      projectsZomeApi
+    )
+
+    expect(installProjectApp).toHaveBeenCalledTimes(1)
+    expect(installProjectApp).toHaveBeenCalledWith(
+      sampleGoodDataExport.projects[0].projectMeta.passphrase
+    )
+
+    expect(createActionHashMapAndImportProjectData).toHaveBeenCalledTimes(1)
+    expect(createActionHashMapAndImportProjectData).toHaveBeenCalledWith(
+      sampleGoodDataExport.projects[0],
+      mockCellIdString,
+      store.dispatch
+    )
+
+    expect(store.dispatch).toHaveBeenCalledTimes(2)
+    expect(store.dispatch).toHaveBeenNthCalledWith(1, {
+      type: 'SIMPLE_CREATE_PROJECT_META',
+      payload: projectMeta,
+      meta: { cellIdString: mockCellIdString },
+    })
+    expect(store.dispatch).toHaveBeenNthCalledWith(2, {
+      type: 'SET_MEMBER',
+      payload: {
+        cellIdString: mockCellIdString,
+        member: {
+          agentPubKey: 'testAgentAddress',
+        },
+      },
+    })
+  })
+})
+
+describe('importProjectData()', () => {
+  it('creates actionHashMaps for all relevant data types', async () => {
+    let projectData = sampleGoodDataExport.projects[0]
+    projectsZomeApi = createProjectsZomeApi(mockAppWs)
+
+    await internalCreateActionHashMapAndImportProjectData(
+      projectData,
+      mockCellIdString,
+      store.dispatch,
+      projectsZomeApi,
+      cloneDataSet
+    )
```

Comment:
since this is a method with return values, the result should be assigned, and asserted